File: Native/Test/Fabric/kl-diagnostics.js
FABRIC = require('Fabric').createClient();

op = FABRIC.DependencyGraph.createOperator( "op" );
console.log( op.getDiagnostics().length );

op.setSourceCode("operator entry( io Scalar input, io Scalar output ) { foo; }");
console.log( op.getDiagnostics().length );
console.log( op.getDiagnostics()[0].desc );

op.setSourceCode("operator entry( io Scalar input, io Scalar output ) { output = 2 * input; }");
console.log( op.getDiagnostics().length );

FABRIC.flush();

File: Native/Test/Python/kl-diagnostics.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric
fabricClient = fabric.createClient()

op = fabricClient.DependencyGraph.createOperator( "op" )
print( len( op.getDiagnostics() ) )

op.setSourceCode("operator entry( io Scalar input, io Scalar output ) { foo; }")
print( len( op.getDiagnostics() ) )
print( op.getDiagnostics()[0]['desc'] )

op.setSourceCode("operator entry( io Scalar input, io Scalar output ) { output = 2 * input; }")
print( len( op.getDiagnostics() ) )

fabricClient.close()
